pkg/store/prometheus.go

```diff
@@ -496,9 +500,51 @@ func (p *PrometheusStore) LabelValues(ctx context.Context, r *storepb.LabelValue
 		return &storepb.LabelValuesResponse{Values: []string{l}}, nil
 	}
 
-	vals, err := p.client.LabelValuesInGRPC(ctx, p.base, r.Label, nil, r.Start, r.End)
-	if err != nil {
-		return nil, err
+	var (
+		sers []map[string]string
+		err  error
+	)
+
+	lvc := false // LabelValuesCall
+	vals := []string{}
+	v := p.promVersion()
+
+	version, err := semver.Parse(v)
+	if err == nil {
+		baseVer, err := semver.Make("2.24.0")
+		if err != nil {
```

Comment:
If this is true(https://github.com/thanos-io/thanos/pull/3940/files#r622405182), then we should not exit early. Then we can remove the error check.

Reply:
> Then we can remove the error check.

Sorry, are you saying about the error check at Line 515? As it is for handling error retuned while initializing the `baseVer`.

Reply:
Yup, if it errors out, we can revert to our original behaviour of - 
> we can still make a series call instead of returning the error

Reply:
Does this look good?
```
version, err1 := semver.Parse(v)

baseVer, err2 := semver.Make("2.24.0")

if err1 == nil && err2 == nil && version.Compare(baseVer) == 1 {

    lvc = true

}

```

Reply:
Apologies for elongating this, but we can simplify things by using this [package](https://pkg.go.dev/golang.org/x/mod/semver#Compare)

I checked their docs and it stated for their [Compare](https://pkg.go.dev/golang.org/x/mod/semver#Compare) function, we can provide two strings, if the promVersion is invalid, the value would be `0`, so we dont have to worry about that. So lets use this - 

```
defaultVersion := "2.24.0"
if len(r.Matchers) == 0 || semver.Compare(version, defaultVersion) == 1 {
```

Our current implementation is making it unnecessarily verbose and by using the above package, we can simplify things 😄

Reply:
Any thoughts on this @yeya24? As you earlier suggested to use a library https://github.com/thanos-io/thanos/pull/3940#discussion_r606446107

Reply:
If we already use the semver library then let's use it here as well.